### PR TITLE
demand_paging: LRU eviction: add missing documentation

### PR DESCRIPTION
**Pull request by npitre**
Document missing eviction algorithm interface functions, and mention LRU.

### CODE REVIEW

**Review by dcpleung — DISMISSED** [review]

**Review by nashif — DISMISSED** [review]

File: doc/kernel/memory_management/demand_paging.rst [hunk 1]
```diff
@@ -118,12 +118,19 @@ Eviction Algorithm
...
 called from the kernel paging code:
 
 * :c:func:`k_mem_paging_eviction_init()` is called to initialize
   the eviction algorithm. This is called at ``POST_KERNEL``.
 
+* :c:func:`k_mem_paging_eviction_add()` is called each time a data page becomes
+  eligible to future eviction.
+
+* :c:func:`k_mem_paging_eviction_remove()` is called when a data page is no
+  longer eligible to eviction. This may happen if given data page gets
```
Comment (peter-mitsis) [inline doc/kernel/memory_management/demand_paging.rst:131]: eligible to eviction -> eligible for eviction   (applies to line 128 as well).

if given -> if the given

File: doc/kernel/memory_management/demand_paging.rst [hunk 2]
```diff
@@ -118,12 +118,19 @@ Eviction Algorithm
...
+  pinned, is unmapped or is about to be evicted.
```
Comment (peter-mitsis) [inline doc/kernel/memory_management/demand_paging.rst:132]: Minor: gets pinned -> becomes pinned
      is unmapped -> becomes unmapped (or "gets unmapped" if you prefer gets to become)

Rationale: This highlights that it occurs at a state transition.

File: doc/kernel/memory_management/demand_paging.rst [hunk 3]
```diff
@@ -133,12 +140,24 @@ called from the kernel paging code:
...
-ranks each data page on whether they have been accessed and modified.
-The selection is based on this ranking.
+There is one additional function which is called by the architecture's memory
+management code to flag data pages when they trigger an access fault:
+:c:func:`k_mem_paging_eviction_accessed()`. This is used by the LRU algorithm
+to requeue "used" pages.
+
+Two eviction algorithms are currently available:
+
+* A NRU (Not-Recently-Used) eviction algorithm has been implemented as a
```
Comment (peter-mitsis) [inline doc/kernel/memory_management/demand_paging.rst:150]: A NRU -> An NRU

(When spoken, NRU begins with a vowel sound)

File: doc/kernel/memory_management/demand_paging.rst [hunk 4]
```diff
@@ -133,12 +140,24 @@ called from the kernel paging code:
...
+  sample. This is a very simple algorithm which ranks each data page on
+  whether they have been accessed and modified. The selection is based on
+  this ranking.
+
+* A LRU (Least-Recently-Used) eviction algorithm is also available. It is
```
Comment (peter-mitsis) [inline doc/kernel/memory_management/demand_paging.rst:155]: A LRU -> An LRU

(When spoken, LRU begins with a vowel sound)

File: doc/kernel/memory_management/demand_paging.rst [hunk 5]
```diff
@@ -133,12 +140,24 @@ called from the kernel paging code:
...
+There is one additional function which is called by the architecture's memory
+management code to flag data pages when they trigger an access fault:
+:c:func:`k_mem_paging_eviction_accessed()`. This is used by the LRU algorithm
+to requeue "used" pages.
+
+Two eviction algorithms are currently available:
+
+* A NRU (Not-Recently-Used) eviction algorithm has been implemented as a
+  sample. This is a very simple algorithm which ranks each data page on
+  whether they have been accessed and modified. The selection is based on
```
Comment (peter-mitsis) [inline doc/kernel/memory_management/demand_paging.rst:152]: Either ...
   they have -> it has
or ...
   each data page -> data pages

(but not both).

File: doc/kernel/memory_management/demand_paging.rst [hunk 6]
```diff
@@ -133,12 +140,24 @@ called from the kernel paging code:
...
+  this ranking.
+
+* A LRU (Least-Recently-Used) eviction algorithm is also available. It is
+  based on a sorted queue of data pages. The LRU code is more complex compared
+  to the NRU code but also way more efficient. This is recommended for
```
Comment (peter-mitsis) [inline doc/kernel/memory_management/demand_paging.rst:157]: way more -> much more / considerably more / significantly more

File: doc/kernel/memory_management/demand_paging.rst [hunk 7]
```diff
@@ -133,12 +140,24 @@ called from the kernel paging code:
...
+  production use.
 
-To implement a new eviction algorithm, the two functions mentioned
+To implement a new eviction algorithm, the 5 functions mentioned
```
Comment (peter-mitsis) [inline doc/kernel/memory_management/demand_paging.rst:160]: Similar to the comment for line 121-- 5 -> five. (Feel free to ignore)

**Review by peter-mitsis — COMMENTED** [review]
Generally looks good.  Just a few nitpicks.

**Comment by npitre** [comment]
Thanks for reviewing my late night bad english
(although not all of it was mine).  ;-)

**Review by dcpleung — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Review by peter-mitsis — APPROVED** [review]

**Pull request merged**